public/app/plugins/datasource/grafana-azure-monitor-datasource/grafanaTemplateVariableFns.ts

```diff
@@ -0,0 +1,263 @@
+        metricDefinition: matchesForQuery.metricNamespaceWithSub[3],
+        resourceName: matchesForQuery.metricNamespaceWithSub[4],
+      };
+      return queryDetails;
+    }
+
+    if (matchesForQuery.metricNames && defaultSubscriptionId) {
+      if (matchesForQuery.metricNames[3].indexOf(',') === -1) {
+        const queryDetails: MetricNamesQuery = {
+          kind: 'MetricNamesQuery',
+          rawQuery,
+          subscription: defaultSubscriptionId,
+          resourceGroup: matchesForQuery.metricNames[1],
+          metricDefinition: matchesForQuery.metricNames[2],
+          resourceName: matchesForQuery.metricNames[3],
+          metricNamespace: matchesForQuery.metricNames[4],
+        };
+        return queryDetails;
+      }
+    }
+
+    if (matchesForQuery.metricNamesWithSub) {
+      const queryDetails: MetricNamesQuery = {
+        kind: 'MetricNamesQuery',
+        rawQuery,
+        subscription: matchesForQuery.metricNamesWithSub[1],
+        resourceGroup: matchesForQuery.metricNamesWithSub[2],
+        metricDefinition: matchesForQuery.metricNamesWithSub[3],
+        resourceName: matchesForQuery.metricNamesWithSub[4],
+        metricNamespace: matchesForQuery.metricNamesWithSub[5],
+      };
+      return queryDetails;
+    }
+
+    if (matchesForQuery.workspacesQuery && defaultSubscriptionId) {
+      const queryDetails: WorkspacesQuery = {
+        kind: 'WorkspacesQuery',
+        rawQuery,
+        subscription: defaultSubscriptionId,
+      };
+      return queryDetails;
+    }
+
+    if (matchesForQuery.workspacesQueryWithSub) {
+      const queryDetails: WorkspacesQuery = {
+        kind: 'WorkspacesQuery',
+        rawQuery,
+        subscription: (matchesForQuery.workspacesQueryWithSub[1] || '').trim(),
+      };
+      return queryDetails;
+    }
+
+    // fallback
+    const queryDetails: SubscriptionsQuery = { kind: 'SubscriptionsQuery', rawQuery };
+    return queryDetails;
+  };
+
+  const query: AzureMonitorQuery = {
+    refId: 'A',
+    queryType: AzureQueryType.GrafanaTemplateVariableFn,
+    grafanaTemplateVariableFn: createGrafanaTemplateVariableDetails(),
+    subscription: defaultSubscriptionId,
+  };
+  return query;
+};
+
+const createLogAnalyticsTemplateVariableQuery = async (
+  rawQuery: string,
+  datasource: DataSource
+): Promise<AzureMonitorQuery> => {
+  const defaultSubscriptionId = datasource.azureMonitorDatasource.defaultSubscriptionId;
+
+  // convert old workspace ids into new resources (same logic as in useMigrations but out of the react-hook land)
+  const defaultWorkspaceId = datasource.azureLogAnalyticsDatasource.getDeprecatedDefaultWorkSpace();
+  let resource = '';
+  if (defaultWorkspaceId) {
+    const isWorkspaceGUID = isGUIDish(defaultWorkspaceId);
+    if (isWorkspaceGUID) {
+      resource = await datasource.resourcePickerData.getResourceURIFromWorkspace(defaultWorkspaceId);
+    } else {
+      resource = defaultWorkspaceId;
+    }
+  }
```

Comment:
Otherwise, I think this should be the first LA workspace returned from the API.

Reply:
(to match the previous behaviour)

Reply:
Ah interesting... because we have to support old queries that relied on the first workspace yeah? ok will take a look at that.